keg/generator.py

```diff
@@ -19,69 +19,98 @@
+    Class for creating a KIWI image description from Keg Templates
 
-    Expects a keg-recipes structure at recipes_root and image_source
-    referring to a directory under images in recipes_root.
+    :param object image_definition: Instance of KegImageDefinition
+    :param str dest_dir: Destination directory
     """
-    # Sanity check image source location
-    if not os.path.exists(os.path.join(recipes_root, 'images', image_source)):
-        errmsg = 'Source directory for {} does not exist.'.format(image_source)
-        if force:
-            log.warning(errmsg)
-        else:
-            raise KegError(errmsg)
+    def __init__(
+        self, image_definition: KegImageDefinition, dest_dir: str
+    ):
+        if not os.path.isdir(dest_dir):
+            raise KegError(
+                'Given destination directory: {target} does not exist'.format(
+                    target=repr(dest_dir)
+                )
+            )
+        self.image_definition = image_definition
+        self.description_schemas = os.path.join(
+            image_definition.recipes_root, 'schemas'
+        )
+        self.dest_dir = dest_dir
+        self.env = Environment(
+            loader=FileSystemLoader(self.description_schemas)
+        )
+        self.env.globals['keg_funcs'] = template_functions
 
-    img = image_definition.ImageDefinition(image_source, recipes_root, data_roots)
-    img.populate()
+        self.image_definition.populate()
 
-    schemas_dir = os.path.join(recipes_root, 'schemas')
-    env = Environment(loader=FileSystemLoader(schemas_dir))
-    env.globals['keg_funcs'] = template_functions
+    def create_kiwi_description(
+        self, markup: str = 'xml', override: bool = False
+    ):
+        """
+        Creates KIWI config.xml from a KegImageDefinition.
 
-    kiwi_templ = env.get_template('{}.kiwi.templ'.format(img['schema']))
-    kiwi_doc = kiwi_templ.render(data=img.get_data())
+        :param bool override:
+            Override destination contents, default is: False
+        """
+        outfile = os.path.join(self.dest_dir, 'config.kiwi')
+        self._validate_outfile(outfile, override)
 
-    outfile = os.path.join(dest_dir, 'config.kiwi')
-    if os.path.exists(outfile) and not force:
-        raise KegError('{} exists, use force to overwrite.'.format(outfile))
-    with open(outfile, 'w') as fd:
-        fd.write(kiwi_doc)
+        kiwi_template = self.env.get_template(
+            '{}.kiwi.templ'.format(self.image_definition.data['schema'])
+        )
+        kiwi_document = kiwi_template.render(
+            data=self.image_definition.data
+        )
+        with open(outfile, 'w') as kiwi_config:
+            kiwi_config.write(kiwi_document)
+        kiwi = KiwiDescription(outfile)
+        if markup == 'xml':
+            kiwi.create_XML_description(outfile)
+        if markup == 'yaml':
+            kiwi.create_YAML_description(outfile)
```

Comment:
I think we  should add an `else:` option to log | print a message informing that that markup is not supported.

Reply:
yes I think it should raise an error in this case

Reply:
done